pytorch_lightning/trainer/training_loop.py

```diff
@@ -71,6 +71,8 @@ def num_optimizers(self):
         return num_optimizers
 
     def on_train_start(self):
+        self.trainer.signal_connector.setup()
```

Comment:
n00b question: why not do setup in on_trainer_init?

Reply:
I put it here because the signal handling / keyboard interrupt currently applies to the fit / test routines. There may be other user code between trainer init and fit. I think we only want to do our graceful shutdown for the training loop, and let user handle all other interrupts outside the Lightning code. Does that make sense?

Reply:
> do you have to repeat this for evaluation/test?

yes, I plan to do a follow up for the trainer.test case. 
I didn't want to mix refactoring into this, so I am separating it.

Reply:
also, this shall favor multiple training with a single Trainer in one scriop/execution, right?

Reply:
yes, the graceful shutdown simply applies to the current .fit() we are in.